src/services/contracts.ts

```diff
@@ -14,24 +14,31 @@ import {
 import type { Signer } from '@ethersproject/abstract-signer';
 import type { JsonRpcProvider } from '@ethersproject/providers';
 import debug from 'debug';
+import { z } from 'zod';
 
 import { HARDHAT_CHAIN_ID, HARDHAT_GANACHE_CHAIN_ID } from 'config/env';
 
 const log = debug('coordinape:contracts');
 
-export const supportedChainIds: number[] =
-  Object.keys(deploymentInfo).map(Number);
+const requiredContracts = [
+  'ApeVaultFactoryBeacon',
+  'ApeRouter',
+  'ApeDistributor',
+];
+
+export const supportedChainIds: number[] = Object.entries(deploymentInfo)
+  .filter(([, contracts]) => requiredContracts.every(c => c in contracts))
+  .map(x => Number(x[0]));
+
+const symbols = ['DAI', 'USDC', 'USDT', 'YFI'] as const;
+export const AssetEnum = z.enum(symbols);
+export type Asset = z.infer<typeof AssetEnum>;
```

Comment:
Would it be easier to just instantiate a string enum and use it as a type?
```
export const Asset = {
  DAI = "DAI",
  USDC = "USDC"
  ...
}
```
Don't see why we need use to zod. Happy to be schooled ofc.